### ISSUE TITLE
error reading server metadata: unexpected end of JSON input

### ISSUE THREAD

**Issue by nullamix** [hashicorp/consul#19329]
Sometimes a consul container is down because server_metadata.json file in the data directory is empty. I don't know why and when this happens.
container log:
```
[WARN]  agent.auto_config: bootstrap = true: do not enable unless necessary
[ERROR] agent: startup error: error="error reading server metadata: unexpected end of JSON input"
[ERROR] agent: startup error: error="error reading server metadata: unexpected end of JSON input"
```

I use latest version of [this image](https://hub.docker.com/r/bitnami/consul/)

**Comment by prasanjitshome** [comment]
Any updates on this ?

**Comment by gwsas** [comment]
This looks like a dupe of https://github.com/hashicorp/consul/issues/19720

**Comment by huikang** [comment]
This fix will be released in the subsequent patch releases of 1.15, 1.16, and 1.17, which should be by end of Jan.

**Comment by nullamix** [comment]
Please let me check again by the end of January. Thanks for your help.
